redbot/core/bot.py

```diff
@@ -149,6 +149,12 @@ def __init__(self, *args, cli_flags=None, bot_dir: Path = Path.cwd(), **kwargs):
         if "command_not_found" not in kwargs:
             kwargs["command_not_found"] = "Command {} not found.\n{}"
 
+        message_cache_size = cli_flags.message_cache_size
+        if cli_flags.no_message_cache:
+            message_cache_size = None
+        kwargs["max_messages"] = message_cache_size
+        self._max_messages = message_cache_size
```

Comment:
While this should remain private, adding a read-only property (in similar way as `bot.uptime`) `bot.max_messages` could be useful for cog creators to know if cache is enabled.
Alternatively, a `True/False` read-only property that just tells cog creator if cache is enabled or not would be enough too.

Reply:
I would prefer if the attribute would not mirror the one in discord.py's kwargs since it could be potentially confusing.

Reply:
For posterity, there was a confusion caused because of my suggestion to use `True/False` as value. If we decided to go that way, it should certainly have a different name than `max_messages` but the approach currently taken in this PR (which is just using the same thing we pass to d.py in kwargs) is fine with `max_messages`.